### PR TITLE
initial CODEOWNERS

### PR DESCRIPTION
**Pull request by spencergilbert**
This is adding some high level owners to some of the code, but intentionally not all of it as I'm not sure where some of those belong. As I didn't set any owners for the default (`*`) we can opt in as needed. 

[About code owners](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners)

### CODE REVIEW

**Review by neuronull — APPROVED** [review]

**Review by davidhuie-dd — APPROVED** [review]

**Review by jszwedko — COMMENTED** [review]
Great! One thing I think could be nice would be to list out the components for each owned directory. For example:

```
src/sinks/amqp/ @StephenWakely # sink_amqp
```

or

```
# sink_amqp
src/sinks/amqp/ @StephenWakely
```

Using a more predictable directory naming and structure could also allow for this without the need for comments.

This would make it easier to parse the file to generate ownership counts.

**Comment by spencergilbert** [comment]
Added components that couldn't be derived directly by the path as a commented CSV

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,87 @@
+lib/vector-core/ @vectordotdev/core-team
+
+src/vrl/ @vectordotdev/processing-team
```
Comment (bruceg) [inline .github/CODEOWNERS:3]: Also `lib/vrl` and `lib/vector-vrl-functions`

**Review by bruceg — CHANGES_REQUESTED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (spencergilbert) [reply .github/CODEOWNERS:3]: 😓 `src/vrl` isn't a thing, should have been `lib`

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+src/sinks/blackhole/ @davidhuie-dd
+src/sinks/clickhouse/ @davidhuie-dd
+src/sinks/console/ @davidhuie-dd
+src/sinks/datadog_archives.rs @neuronull
+src/sinks/datadog_events/ @neuronull
+src/sinks/datadog_logs/ @neuronull
+src/sinks/datadog_metrics/ @neuronull
+src/sinks/datadog_traces/ @neuronull
+src/sinks/elasticsearch/ @spencergilbert
+src/sinks/file/ @spencergilbert
```
[the remark this answers is not in the document]
Comment (spencergilbert) [reply .github/CODEOWNERS:34]: Resolved in c71059afb

File: .github/CODEOWNERS (hunk 1 above)
Comment (spencergilbert) [reply .github/CODEOWNERS]: Resolved in c71059afb

**Review by bruceg — APPROVED** [review]

**Review by neuronull — APPROVED** [review]

**Pull request merged**